types/airbnb-prop-types/package.json

```
@@ -10,7 +10,7 @@
     },
     "devDependencies": {
         "@types/airbnb-prop-types": "workspace:.",
-        "@types/react": "*"
+        "@types/react": "^15"
```

Hm, upstream says `^16.0.0-alpha`, which should match `^16`; are you seeing failures in newer versions or is this just something you noticed?

---

I was under the impression `^16.0.0-alpha` doesn't match stable releases but it does
```suggestion
        "@types/react": "^16"
```

---

I was under the impression `^16.0.0-alpha` doesn't match stable releases but it does according to https://semver.npmjs.com/
```suggestion
        "@types/react": "^16"
```

---

Thanks, but I am still curious as to what this is needed for; is it just to lower how many tests are run or something?

---

Because that's what's happening at runtime. This package is not compatible with React 17 and beyond so why should it pull in those types?

---

Sure, though loads of react packages declare max compat with some react version and actually support higher with users using it anyway since peer deps can be ignored most of the time. (e.g. react-dnd-whatever-it's-called). But, reading https://github.com/airbnb/prop-types/issues/73, it's probably fine to be stricter here.

---

> Sure, though loads of react packages declare max compat with some react version

And I'd prefer types do not do that: https://github.com/microsoft/DefinitelyTyped-tools/issues/433

It used to be something can't control unless we pin the version to an older major but maybe that changed?